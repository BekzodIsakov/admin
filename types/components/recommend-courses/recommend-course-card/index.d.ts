export * from "./recommend-course-card.component";
//# sourceMappingURL=index.d.ts.map