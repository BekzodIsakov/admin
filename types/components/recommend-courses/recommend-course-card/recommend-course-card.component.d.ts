export const RecommendCourseCard: any;
//# sourceMappingURL=recommend-course-card.component.d.ts.map