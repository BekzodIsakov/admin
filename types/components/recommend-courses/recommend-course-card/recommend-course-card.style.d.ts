export const RecommendedCourseCard: any;
export const RecommendCourseImage: any;
export const RecommendCourseTitle: any;
export const RecommendedCourseText: any;
export const RecommendCourseCardLink: any;
//# sourceMappingURL=recommend-course-card.style.d.ts.map