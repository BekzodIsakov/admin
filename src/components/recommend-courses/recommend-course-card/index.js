export * from './recommend-course-card.component';
