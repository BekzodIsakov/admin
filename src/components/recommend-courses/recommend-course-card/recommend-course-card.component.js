import { hoc } from '@utils/hoc';
import { SvgIcon } from '@components/atoms';

import {
  RecommendedCourseCard,
  RecommendCourseImage,
  RecommendCourseTitle,
  RecommendedCourseText,
  RecommendCourseCardLink
} from './recommend-course-card.style';
import { config } from 'src/config';

const useRecommendCourseCardProps = () => {};

export const RecommendCourseCard = hoc(
  useRecommendCourseCardProps,
  ({ cardTitle, cardText, cardToLink, cardImg }) => {
    return (
      <RecommendedCourseCard>
        <RecommendCourseImage
          src={
            cardImg ??
            'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/1200px-Image_created_with_a_mobile_phone.png'
          }
          data-testid='recommended-course-img'
        />

        <RecommendCourseTitle data-test-id='recommended-course-title'>
          {cardTitle}
        </RecommendCourseTitle>

        <RecommendedCourseText data-testid='recommended-course-text'>
          {cardText.slice(0, 75) + '...'}
        </RecommendedCourseText>

        <RecommendCourseCardLink
          href={`${config.landingUrl}/courses/${cardToLink}`}
          target='_blank'
          data-testid='recommended-courseLink'
        >
          Batafsil
          <SvgIcon
            size='12px'
            iconColor='#BA8D5B'
            $stroke='s'
            iconName='arrow_right-icon'
          />
        </RecommendCourseCardLink>
      </RecommendedCourseCard>
    );
  },
  'RecommendCourseCard'
);
