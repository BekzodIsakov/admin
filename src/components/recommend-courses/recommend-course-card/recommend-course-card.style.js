import styled from 'styled-components';

export const RecommendedCourseCard = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #fff;
  width: 32%;
`;

export const RecommendCourseImage = styled.img`
  width: 100%;
  height: 158px;
  border-radius: 4px;
  object-fit: cover;
  margin-bottom: 16px;
  object-position: right center;
`;

export const RecommendCourseTitle = styled.h4`
  font-style: normal;
  font-weight: bold;
  font-size: 18px;
  line-height: 22px;
  letter-spacing: 0;
  color: #000000;
  margin-bottom: 6px;
`;

export const RecommendedCourseText = styled.p`
  flex-grow: 1;
  font-style: normal;
  font-weight: normal;
  font-size: 1.4rem;
  line-height: 16px;
  letter-spacing: -0.02em;
  color: rgba(0, 0, 0, 0.5);
`;

export const RecommendCourseCardLink = styled.a`
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  line-height: 18px;
  letter-spacing: -0.02em;
  display: inline-block;
  text-decoration: none;
  padding: 19px 0;
  color: #ba8d5b;

  svg {
    margin-left: 12px;
    transition: all ease 0.3s;
  }

  &:hover svg {
    margin-left: 18px;
  }
`;
