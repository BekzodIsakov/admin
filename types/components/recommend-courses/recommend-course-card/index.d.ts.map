{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/recommend-courses/recommend-course-card/index.js"],"names":[],"mappings":""}