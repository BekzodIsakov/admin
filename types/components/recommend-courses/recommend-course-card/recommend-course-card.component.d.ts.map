{"version":3,"file":"recommend-course-card.component.d.ts","sourceRoot":"","sources":["../../../../src/components/recommend-courses/recommend-course-card/recommend-course-card.component.js"],"names":[],"mappings":"AAcA,sCAgCE"}