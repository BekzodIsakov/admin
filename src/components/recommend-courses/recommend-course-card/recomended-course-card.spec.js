import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';

import { RecommendCourseCard } from './recommend-course-card.component';

const setup = ({ cardTitle, cardText, cardToLink, cardImg }) => {
  return customRender(
    <RecommendCourseCard
      cardTitle={cardTitle}
      cardText={cardText}
      cardToLink={cardToLink}
      cardImg={cardImg}
    />
  );
};

test('Recomended course test props', () => {
  const { asFragment } = setup({
    cardTitle: 'cardTitle',
    cardText: 'cardText',
    cardImg: 'cardImg'
  });
  expect(asFragment).toMatchSnapshot();
});

test('render recomended course cards', () => {
  setup({ cardTitle: 'cardTitle', cardText: 'cardText', cardToLink: '/link' });
  const recomendedCoursesTitle = screen.getByRole('heading', {
    name: 'cardTitle'
  });
  const recommendedCoursesText = screen.getByTestId('recommended-course-text');
  const recommendedCourseLink = screen.getByTestId('recommended-courseLink');
  expect(
    recomendedCoursesTitle && recommendedCoursesText && recommendedCourseLink
  ).toBeTruthy();
});
