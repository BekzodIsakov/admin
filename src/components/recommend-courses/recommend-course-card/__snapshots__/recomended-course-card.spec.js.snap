// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recomended course test props 1`] = `[Function]`;
